Add contributor details to People page

ShinichiShi opened the pull request:
## Description
This PR completes the People page by adding the contributors details into it. It provides a overall view of Circuitverse contributors along with each contributor details. Filtering based on points is also possible. 
## Related Issue
Fixes #2 

## Type of change
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Documentation

## Checklist
- [x] Code follows project style
- [x] Tested locally
- [x] No unnecessary files added
- [x] PR title is clear and descriptive

## Screenshots (if applicable)
[Screencast from 2025-12-27 02-21-20.webm](https://github.com/user-attachments/assets/f34bca84-a0ff-4d42-a0dc-98f7accb7fed)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * People page rebuilt as a client-driven leaderboard with live fetching, loading/error states and updated timestamp
  * Search, filters, sorting and grid/list view controls to browse contributors
  * Clickable contributor cards with a detailed profile view, activity metrics, recent contributions and a GitHub heatmap
  * Team sections for Core Team and Alumni, aggregated stats and top contributors
  * New reusable UI primitives and consolidated people component exports

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

naman79820 commented:
Hey @ShinichiShi you did a really really great job brother. Everything looks excellent 🤩.  it's fully responsive and working perfectly.
Just a small request could you please update the colors to something more vibrant? You can choose whatever colors you think look best here .
<img width="980" height="161" alt="image" src="https://github.com/user-attachments/assets/1668dee6-a5ba-4db5-ba49-a685cefc9e42" />

ShinichiShi commented:
Hey @naman79820 @Atharva7126 i have made the fiollowing few changes:
- added the core team and alumni team take from the /about page of cv
- removed filtering as its already present in leaderboard
- added github heatmap for the individual contributor page

further improvements must be made, but can i raise separate issues for them in order to make it easier to review PR?

naman79820 commented on app/api/people/route.ts:
```diff
@@ -0,0 +1,75 @@
+              entry.username.includes('renovate') ||
+              entry.username.includes('github-actions')) {
+            continue;
+          }
```
Hey @ShinichiShi regarding this bot issue, can you use same logic which circuit verse about page does
Ref:- https://github.com/naman79820/circuitverse-leaderboard/commit/66168efe3b6a4d36d0d290ebd6a7945e481fd297#diff-c4ce6fe99bbfe74d3e9237842e1cf315c9c26fd7d66ffd145337585a51eb66ae

ShinichiShi commented:
> great work so far @ShinichiShi other things can be done in other issues, one last thing, there are some protentional issues created by coderabbit, can you resolve that plss

these issues have been resolved
